chia/wallet/puzzles/dao_spend_p2_singleton_v2.clsp

```diff
@@ -104,6 +105,26 @@
     )
   )
 
+  (defun reverse (lst output)
+    (if lst
+        (reverse (r lst) (c (f lst) output))
+        output
+    )
+  )
+
+  (defun dedupe ((@ lst ((@ first (parent amount)) . rest)) ids output)
+    (if lst
+        (let ((id (sha256 parent amount)))
+          (if (in id ids)
+              (dedupe rest ids output)
+              (dedupe rest (c id ids) (c first output))
```

Comment:
if you do `(c first (dedupe rest (c id ids)))` you can get rid of the output field and the reverse operation